### PR TITLE
MON-106121-pull-wss-add-gorgone-parameter-to-get-vault-credentials

### PR DESCRIPTION
**Pull request by Evan-Adam**
## Description
move the centreonvault.pm file made in centreon-plugin to the generic library stored here.
splitted the package in 2, centreon-perl-libs-common will be installed by everyone, centreon-perl-libs will only be installed by webapp (it contain mainly logic for script about dashboard, trap management and reporting)

**Fixes** # mon-106121

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [X] master

<h2> How this pull request can be tested ? </h2>
install both package, they should not have any error.
The library inside should all be callable by using a command of the form 
```
perl -e 'use strict ; use centreon::common::logger; print "ok\n" '
```

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/lib/perl/t/common/logger.t [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+use Test2::V0;
+use FindBin;
+use lib "$FindBin::Bin/../../";
+use centreon::common::logger;
```
[the remark this answers is not in the document]
Comment (Evan-Adam) [reply centreon/lib/perl/t/common/logger.t:8]: interresting but not a core module yet

File: centreon/lib/perl/t/common/logger.t [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        local *STDOUT;
+        open STDOUT, ">", \$out;
+        $logger->writeLogInfo($logExemple);
+    };
+    my $log = check_log_format(log => $out, severity => "INFO", pid => $options{pid});
+
+    is($log, $logExemple, "log is the same as what we sent.");
+
+    print "    written to original STDOUT : $out";
+}
```
[the remark this answers is not in the document]
Comment (Evan-Adam) [reply centreon/lib/perl/t/common/logger.t:45]: @coderabbitai give me the documentation of capture_stdout  function

File: centreon/lib/perl/t/common/logger.t [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        open STDOUT, ">", \$out;
+        # for now we just check the function exist, we should check the log is written correctly in the future.
+        $logger->writeLogDebug($logExemple);
+        $logger->writeLogInfo($logExemple);
+        $logger->writeLogNotice($logExemple);
+        $logger->writeLogWarning($logExemple);
+        $logger->writeLogError($logExemple);
+        eval {$logger->writeLogFatal($logExemple); }; # for now we just ignore the die()
+    };
+}
```
[the remark this answers is not in the document]
Comment (Evan-Adam) [reply centreon/lib/perl/t/common/logger.t:83]: @coderabbitai yes please make an implementation

File: centreon/lib/perl/centreon/common/centreonvault.pm [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+    if ( !defined($self->{logger}) ) {
+        die "FATAL: No logger given to the constructor. Centreonvault cannot be used.";
+    }
+    if ( !defined($self->{config_file})) {
+        $self->{logger}->writeLogError("No config file given to the constructor. Centreonvault cannot be used.");
+        return undef;
+    }
+    if ( ! -f $self->{config_file} and ref $self->{config_file} ne ref {}) {
+        $self->{logger}->writeLogError("The given configuration file " . $self->{config_file}
+            . " does not exist. Centreonvault cannot be used.");
```
Comment (omercier) [inline centreon/lib/perl/centreon/common/centreonvault.pm:93]: After some user feedback, I made the error message [less alarming](https://github.com/centreon/centreon-plugins/blob/39d6d4bbf96c9b99c229896e9b98e8c19b2704ed/connectors/vmware/src/centreon/script/centreonvault.pm#L89). If you feel like doing so...

File: .github/workflows/perl-unit-tests.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            image: unit-tests-bookworm
+            distrib: bookworm
+          - package_extension: deb
+            image: unit-tests-jammy
+            distrib: jammy
+    runs-on: ${{ matrix.runner_name }}
+    container:
+      image: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/${{ matrix.image }}
+      credentials:
+        username: ${{ secrets.DOCKER_REGISTRY_ID }}
```
Comment (tuntoja) [inline .github/workflows/perl-unit-tests.yml:57]:
```suggestion
        username: ${{ secrets.HARBOR_CENTREON_PULL_USERNAME }}
```

File: .github/workflows/perl-unit-tests.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
```
Comment (tuntoja) [inline .github/workflows/perl-unit-tests.yml:58]:
```suggestion
        password: ${{ secrets.HARBOR_CENTREON_PULL_TOKEN }}
```

**Review by omercier — APPROVED** [review]

**Review by omercier — APPROVED** [review]

File: centreon/lib/perl/t/common/logger.t (hunk 2 above)
[the remark this answers is not in the document]
Comment (Evan-Adam) [reply centreon/lib/perl/t/common/logger.t]: This function don't seem to exist, give me the link to the documentation or your source about that please.

File: .github/workflows/perl-unit-tests.yml (hunk 5 above)
Comment (Evan-Adam) [reply .github/workflows/perl-unit-tests.yml]: Done

File: centreon/lib/perl/t/common/logger.t (hunk 2 above)
[the remark this answers is not in the document]
Comment (Evan-Adam) [reply centreon/lib/perl/t/common/logger.t]: @coderabbitai This function don't seem to exist, give me the link to the documentation or your source about that please

**Pull request closed without merging**